csharp/src/Ice/IncomingRequestFrame.cs

```diff
@@ -0,0 +1,45 @@
+//
+// Copyright (c) ZeroC, Inc. All rights reserved.
+//
+
+using System;
+using System.Collections.Generic;
+
+namespace Ice
+{
+    using Context = Dictionary<string, string>;
+
+    /// <summary>Represents a request protocol frame received by the application.</summary>
+    // TODO: IncomingRequestFrame should derive from InputStream
+    public sealed class IncomingRequestFrame
```

Comment:
This is class is not yet used, what is the plan finish this in a separate PR?

Reply:
It's actually used as a helper in IceLocatorDiscovery/PluginI.cs and test/Ice/echo/BlobjectI.cs.
And yes, a future PR should update the dispatch code to create it automatically.